README.md

```diff
@@ -19,27 +19,75 @@ kubectl create ns appmesh-system
 
 ```sh
-helm upgrade -i appmesh-controller eks/appmesh-controller \
+helm upgrade -i appmesh-manager eks/appmesh-manager \
 --namespace appmesh-system
 ```
 
-Install the App Mesh admission controller:
+The new AppMesh controller v1.0.0 is backwards-incompatible with older AppMesh controllers(<v1.0.0) and v1beta1
+AppMesh CRDs. If you have an old AppMesh controller/injector or v1beta1 AppMesh CRDs installed in your cluster, please
+follow the steps in the relevant upgrade sections below
+
+#### Upgrade without preserving old AppMesh resources
 
 ```sh
-helm upgrade -i appmesh-inject eks/appmesh-inject \
---namespace appmesh-system \
---set mesh.create=true \
---set mesh.name=global
+# Keep old AppMesh controller running, it is responsible to cleanup AppMesh resources in AWS
+# Delete all existing AppMesh CRs
+kubectl delete virtualservices --all --all-namespaces
+kubectl delete virtualnodes --all --all-namespaces
+kubectl delete meshes --all --all-namespaces
+
+# Delete all existing AppMesh CRDs
+kubectl delete customresourcedefinition/virtualservices.appmesh.k8s.aws
+kubectl delete customresourcedefinition/virtualnodes.appmesh.k8s.aws
+kubectl delete customresourcedefinition/meshes.appmesh.k8s.aws
+# Note: If a CRD stuck in deletion, it means there still exists some AppMesh CRs, please check and delete them.
+
+# Delete AppMesh controller
+helm delete appmesh-controller -n appmesh-system
+
+# Delete AppMesh injector
+helm delete appmesh-inject -n appmesh-system
 ```
+After cleanup, you can proceed with the installation steps described above
+
+#### Upgrade preserving old AppMesh resources
 
-If you've installed the App Mesh controllers with scripts, you can switch to Helm by
-removing the controllers with:
+```sh
+# Save manifests of all existing AppMesh CRs:
+kubectl get virtualservices --all-namespaces -o yaml > virtualservices.yaml
+kubectl get virtualnodes --all-namespaces -o yaml > virtualnodes.yaml
+kubectl get meshes --all-namespaces -o yaml > meshes.yaml
+
+# Delete AppMesh controller, so it won’t clean up AppMesh resources in AWS while we deleting AppMesh CRs later.
+helm delete appmesh-controller -n appmesh-system
+
+# Delete AppMesh injector.
+helm delete appmesh-inject -n appmesh-system
+
+# Remove finalizers from all existing AppMesh CRs.  (Otherwise, you won’t be able to delete them)
+# Delete all existing AppMesh CRs:
+kubectl delete virtualservices --all --all-namespaces
+kubectl delete virtualnodes --all --all-namespaces
+kubectl delete meshes --all --all-namespaces
+
+# Delete all existing AppMesh CRDs.
+kubectl delete customresourcedefinition/virtualservices.appmesh.k8s.aws
+kubectl delete customresourcedefinition/virtualnodes.appmesh.k8s.aws
+kubectl delete customresourcedefinition/meshes.appmesh.k8s.aws
+# Note: If CRDs are stuck in deletion, it means there still exists some AppMesh CRs, please check and delete them.
+```
+Translate the saved old YAML manifests using v1beta1 AppMesh CRD into v1beta2 AppMesh CRD format.
```

Comment:
Maybe add a link to v1beta2 format or some samples?

Reply:
I will add link to the https://github.com/aws/aws-app-mesh-examples and API doc.

Reply:
We will also update this after we have the upgrade tools available.